iCubGenova09/hardware/inertials/head-inertial.xml

```diff
@@ -43,7 +43,7 @@
             </group>
 
             <group name="SETTINGS">
-                <param name="acquisitionRate">      10      </param>
+                <param name="acquisitionRate">      5      </param>
```

Comment:
If I got it right, it seems that `10` is the fastest allowed rate as per https://github.com/robotology/icub-main/blob/master/src/libraries/icubmod/embObjInertials/embObjInertials.cpp#L527-L538.

Have you seen the following warning message popping up?

```
embObjInertials::sendConfig2MTBboards() has detected a wrong acquisition rate =...
```

Reply:
Note that this device is of type `embObjIMU`, so I am not sure if any limit like that exists.

Reply:
> If I got it right, it seems that `10` is the fastest allowed rate as per https://github.com/robotology/icub-main/blob/master/src/libraries/icubmod/embObjInertials/embObjInertials.cpp#L527-L538.
> 
> Have you seen the following warning message popping up?
> 
> ```
> embObjInertials::sendConfig2MTBboards() has detected a wrong acquisition rate =...
> ```

The 100 Hz frequency (acquisitionRate =  10 ms) is the maximum frequency that we can squeeze from the IMU Bosch BNO055. And that is due to HW limitation, in particular to the way the chip replies over the used I2C bus. We thoroughly explored this limitation some time ago.
  
So, we should correct the value to be 10.

On the other hand, in [here](https://github.com/ami-iit/robots-configuration/pull/20#issuecomment-1261268881) the intention is to
> - increasing the frequency of IMUs to 100Hz

Reply:
>Note that this device is of type embObjIMU, so I am not sure if any limit like that exists.

Regarding this, the function in question is then at https://github.com/robotology/icub-main/blob/26c74355c9d6463762352f41f8f1a1df593844ca/src/libraries/icubmod/embObjIMU/eo_imu_privData.cpp#L314-L340.

Reply:
> Note that this device is of type `embObjIMU`, so I am not sure if any limit like that exists.

@traversaro: thanks for it. I checked and device `embObjIMU` does not check vs incorrect values of `acquisitionRate` as the original device `embObjInertials` did. Maybe in the migration towards the new device something was missed.

But the FW indeed does the check and clips values in range [5, 200] ms. The lower possible value is 5 because in some cases (e.g., only one or two values retrieved) 5 ms can be OK. But again, 10 ms is the recommended lower value that we should use.